source/Octopus.Tentacle.Tests.Integration/Support/PollingTentacleBuilder.cs

```diff
@@ -2,239 +2,56 @@
-                        {
-                            if (s.Contains("Agent will not listen") || s.Contains("Agent listening on"))
-                            {
-                                hasTentacleStarted.Set();
-                            }
-                        }, cancellationToken);
-                }
-                catch (Exception e)
-                {
-                    TestContext.WriteLine(e);
-                    throw;
-                }
-            }, cancellationToken);
-
-            await Task.WhenAny(runningTentacle, WaitHandleAsyncFactory.FromWaitHandle(hasTentacleStarted.WaitHandle, TimeSpan.FromMinutes(1)));
-
-            // Will throw.
-            if (runningTentacle.IsCompleted) await runningTentacle;
-
-            if (!hasTentacleStarted.IsSet)
-            {
-                throw new Exception("Tentacle did not appear to start correctly");
-            }
+            CreateInstance(tentacleExe, configFilePath, instanceName, tempDirectory, cancellationToken);
+            AddCertificateToTentacle(tentacleExe, instanceName, CertificatePfxPath, tempDirectory, cancellationToken);
+            ConfigureTentacleToPollOctopusServer(configFilePath, subscriptionId);
 
-            return runningTentacle;
+            return await StartTentacle(
+                subscriptionId,
+                tentacleExe,
+                instanceName,
+                tempDirectory,
+                TentacleThumbprint,
+                cancellationToken);
         }
 
-        private void ConfigureTentacleToPollOctopusServer(string configFilePath, int octopusHalibutPort, string octopusThumbprint, Uri tentaclePollSubscriptionId)
+        private void ConfigureTentacleToPollOctopusServer(string configFilePath, Uri subscriptionId)
         {
-            var startUpConfigFileInstanceRequest = new StartUpConfigFileInstanceRequest(configFilePath);
-            using var container = new Program(new string[] { }).BuildContainer(startUpConfigFileInstanceRequest);
+            var writableTentacleConfiguration = GetWritableTentacleConfiguration(configFilePath);
 
-            var writableTentacleConfiguration = container.Resolve<IWritableTentacleConfiguration>();
-            writableTentacleConfiguration.AddOrUpdateTrustedOctopusServer(new OctopusServerConfiguration(octopusThumbprint)
+            writableTentacleConfiguration.AddOrUpdateTrustedOctopusServer(new OctopusServerConfiguration(ServerThumbprint)
             {
-                Address = new Uri("https://localhost:" + octopusHalibutPort),
+                Address = new Uri("https://localhost:" + pollingPort),
                 CommunicationStyle = CommunicationStyle.TentacleActive,
-                SubscriptionId = tentaclePollSubscriptionId.ToString()
+                SubscriptionId = subscriptionId.ToString()
             });
-            writableTentacleConfiguration.SetApplicationDirectory(Path.Combine(new DirectoryInfo(configFilePath).Parent.FullName, "appdir"));
```

Comment:
@acodrington bug here I think with me removing this line

Reply:
Thanks, I didn't notice that line had been removed when I read through the changes 🤦🏻